bgpd/bgp_route.c

```diff
@@ -2354,12 +2378,15 @@ void bgp_best_selection(struct bgp *bgp, struct bgp_dest *dest,
 	/* Check old selected route and new selected route. */
 	old_select = NULL;
 	new_select = NULL;
+
 	for (pi = bgp_dest_get_bgp_path_info(dest);
 	     (pi != NULL) && (nextpi = pi->next, 1); pi = nextpi) {
 		enum bgp_path_selection_reason reason;
 
-		if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED))
+		if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED)) {
 			old_select = pi;
+			old_select_valid = true;
```

Comment:
I believe old_select_valid needs to only be set if the BGP_PATH_ATTR_CHANGED flag is not true.

Reply:
will check this scenario